src/services/project-member-service.ts

```diff
@@ -107,40 +112,41 @@ export const getAllByFilters = async (
       ],
     })
     userIds = users.map((user) => user.id)
+    if (userIds.length > 0) {
+      Object.assign(filter, {
+        user_id: {
+          in: userIds,
+        },
+      })
+    } else {
+      Object.assign(filter, {
+        user_id: null,
+      })
+    }
   }
 
   if (project_name !== undefined) {
     const projects = await ProjectRepository.getAllByName(project_name)
     projectIds = projects.map((project) => project.id)
-  }
-
-  const startDate = new Date(start_date)
-  const endDate = new Date(end_date)
-
-  const filter = {
-    user_id,
-  }
-
-  if (userIds.length > 0) {
-    Object.assign(filter, {
-      user_id: {
-        in: userIds,
-      },
-    })
-  }
-
-  if (projectIds.length > 0) {
-    Object.assign(filter, {
-      project_id: {
-        in: projectIds,
-      },
-    })
+    if (projectIds.length > 0) {
+      Object.assign(filter, {
+        project_id: {
+          in: projectIds,
+        },
+      })
+    } else {
+      Object.assign(filter, {
+        project_id: null,
+      })
+    }
   }
 
   if (role !== undefined) {
-    Object.assign(filter, {
-      project_role_id: parseInt(role),
-    })
+    if (role.trim().length > 0) {
```

Comment:
do we need trim here?

Reply:
Ah, will be changing the empty string for this line `const projectRole = role === "all" ? "" : role` to undefined instead. Thanks!